cg/meta/delivery/delivery.py

```diff
@@ -0,0 +1,170 @@
+            sample_delivery_files: list[DeliveryFile] = (
+                self.get_analysis_sample_delivery_files_by_sample(case=case, sample=sample)
+            )
+            delivery_files.extend(sample_delivery_files)
+        return delivery_files
+
+    def get_analysis_sample_delivery_files_by_sample(
+        self, case: Case, sample: Sample
+    ) -> list[DeliveryFile]:
+        """Return a list of analysis files to be delivered for a specific sample."""
+        sample_tags: list[set[str]] = self.get_analysis_sample_tags_for_workflow(case.data_analysis)
+        analysis_sample_tags = [tag | {sample.internal_id} for tag in sample_tags]
+        sample_files: list[File] = (
+            self.housekeeper_api.get_files_from_latest_version_by_list_of_tags(
+                bundle_name=case.internal_id, tags=analysis_sample_tags
+            )
+        )
+        delivery_files: list[DeliveryFile] = self.convert_files_to_delivery_files(
+            files=sample_files,
+            case=case,
+            source_id=sample.internal_id,
+            destination_id=sample.name,
+            analysis_sample_files=True,
+        )
+        return delivery_files
+
+    def convert_files_to_delivery_files(
+        self,
+        files: list[File],
+        case: Case,
+        source_id: str,
+        destination_id: str,
+        analysis_sample_files: bool = False,
+    ) -> list[DeliveryFile]:
+        """Return a delivery file model given a list of housekeeper files."""
+        delivery_files: list[DeliveryFile] = []
+        destination_path: Path = Path(
+            self.delivery_path, case.customer.internal_id, INBOX_NAME, case.latest_ticket
+        )
+        subfolder: Path = Path(destination_id)
+        if analysis_sample_files:
+            subfolder: Path = Path(case.name, destination_id)
+        for file in files:
+            destination_file_name: str = Path(file.full_path).name.replace(
+                source_id, destination_id
+            )
+            destination_path: Path = Path(destination_path, subfolder, destination_file_name)
+            delivery_file = DeliveryFile(
+                source_path=Path(file.full_path), destination_path=destination_path
+            )
+            delivery_files.append(delivery_file)
+        return delivery_files
+
+    @staticmethod
+    def get_analysis_case_tags_for_workflow(workflow: Workflow) -> list[set[str]]:
+        return PIPELINE_ANALYSIS_TAG_MAP[workflow]["case_tags"]
+
+    @staticmethod
+    def get_analysis_sample_tags_for_workflow(workflow: Workflow) -> list[set[str]]:
+        return PIPELINE_ANALYSIS_TAG_MAP[workflow]["sample_tags"]
+
+    @staticmethod
+    def is_sample_deliverable(sample: Sample, force: bool = False) -> bool:
+        """Return whether the sample is deliverable or not."""
+        is_external: bool = sample.application_version.application.is_external
+        qc_pass: bool = sample.sequencing_qc
+        return is_external or qc_pass or force
```

Comment:
```suggestion
        return any(is_external or qc_pass or force)
```
Is this the same?

Reply:
Any expects an iterable, so it would be the same if we define it as `any([is_external, qc_pass, ...])`. But it just adds more complexity